### DISCUSSION TITLE
Callback to server on company or part crud

### DISCUSSION THREAD

**Discussion by bryanlanghendries** [inventree/InvenTree#7225] [Q&A] [answered]
I want my server to get 'notified' if something gets created, updated or deleted. If I create a company on the inventree frontend I want to be able to get that information on my server so I can do something with that data.

There were several solutions I came up with but I was wondering if there is a better way to do this.
1. Webhooks: I created a webhook, but don't really know how to connect this to certain actions on the frontend and how to receive this webhook.
2. Maybe through notifications as I can get the notifications through the api.
3. Polling: Just looping through every company to look if something changed, by maybe keeping a boolean in the notes. (seems heavy)
4. Plugins: Don't really know where to begin with this.

**Comment by matmair** [comment, marked as answer]
1. Currently implemented webhooks are to be managed by plugins and only for receiving - there is a FR for sending: https://github.com/inventree/InvenTree/issues/2038
2. That is not a good idea, notifications might change at any point and are not structured for M2M communications
3. Seems like a good way to get some usage on the server - would not recommend that
4. Here is an [overview](https://docs.inventree.org/en/stable/extend/plugins/) of plugin development. You could use an [event](https://docs.inventree.org/en/stable/extend/plugins/event/) mixin.

If you need support in developing a specific plugin to integrate the service you want to connect feel free to reach out, my mail is in my profile.

**Reply by bryanlanghendries** [reply]
Thank you for responding so quickly I will look into the plugin tomorrow. I might send you a message though! :)
